dimitriosv/week3/app/app.jsx

```diff
@@ -0,0 +1,123 @@
+import List from './List.jsx';
+
+    var App = React.createClass({
+        getInitialState: function() {
+            return {  //This state can be changed to empty of course
+                    totalLists:2,
+                    totalMessages:4,
+                    inputListName:"",
+                    lists: [{listId: 1, listName: "first list1"},{listId: 2, listName: "second list2"}],
+                    allMessages: [  {messageId:1, belongsToList: 1, text: "testmessage1", isArchived: false},
+                                    {messageId:2, belongsToList: 1, text: "testmessage2", isArchived: false},
+                                    {messageId:3, belongsToList: 1, text: "testmessage3", isArchived: false},
+                                    {messageId:4, belongsToList: 2, text: "testmessage4", isArchived: false}]
+                };
+        },
+        eachList: function(list, i) {  
+        return (
+                <List key={i}
+                    index={i}
+                    listName={list.listName}
+                    listId={list.listId}
+                    allLists={this.state.lists}
+                    addMessageParent={this.addMessageParent}
+                    deleteMessageParent={this.deleteMessageParent}
+                    toggleArchiveMessageParent={this.toggleArchiveMessageParent}
+                    allMessages={this.state.allMessages}
+                >{list}</List>
+            );
+        },
+        createList: function(evt) {
+        if (this.state.inputListName=="") {
+            alert("Cannot add list with no name!")
+        } else {
+            var newlistId=this.state.totalLists
+            newlistId++; 
+            var newList = {
+                listId: newlistId,
+                listName: this.state.inputListName
+            };
+         
+            //var lists=this.state.lists; //WRONG! THIS DOES NOT CREATE A NEW INSTANCE
+            var lists =Array.from(this.state.lists);   //this creates a new instance
+
+            lists.push(newList)   
+             
+            this.setState({
+                  lists: lists,
+                  totalLists: this.state.totalLists+1
+            });
+        }
+             
+            
+        },
+        addMessageParent: function(listId,MessageToAdd) {
+
+            var newMessageId=this.state.totalMessages
+            newMessageId++; 
+
+
+            var newMessage = {
+                messageId:newMessageId,
+                belongsToList: listId,
+                text: MessageToAdd,
+                isArchived: false
+            };
+         
+            var allMessages=this.state.allMessages;
+             
+            allMessages.push(newMessage)
```

Comment:
As I wrote [last week](https://github.com/e-conomic/client-side-application-course/pull/45#discussion_r53710037). You should never mutate state directly. 

`allMessages` still points to the array instance in `this.state`